import/xsl/dspace-xoai.xsl

```diff
@@ -0,0 +1,164 @@
+                  <xsl:if test="position()=1">
+                     <field name="author_sort">
+                        <xsl:value-of select="normalize-space()"/>
+                     </field>
+                  </xsl:if>
+               </xsl:if>
+            </xsl:for-each>
+            <!-- CO AUTHOR -->
+	    <xsl:for-each select="//*[@name='contributor']/*[@name='advisor']/*[@name='none']/*[@name='value']">
+               <field name="author2">
+
+                  <xsl:value-of select="normalize-space(.)"/>
+               </field>
+            </xsl:for-each>
+            <!-- PUBLISHDATE -->
+            <xsl:if test="//*[@name='date']">
+               <field name="publishDate">
+                  <xsl:value-of select="substring(//*[@name='date']/*[@name='accessioned']/*[@name='none']/*[@name='value'], 1, 4)"/>
+               </field>
+               <field name="publishDateSort">
+                  <xsl:value-of select="substring(//*[@name='date']/*[@name='accessioned']/*[@name='none']/*[@name='value'], 1, 4)"/>
+               </field>
+            </xsl:if>
+            <!-- Publisher -->
+            <xsl:if test="//*[@name='publisher']">
+               <field name="publisher">
+		   <xsl:value-of select="//*[@name='publisher']/*/*[@name='value']"/> 
+		   <xsl:if test="//*[@name='dc']/*[@name='relation']/*[@name='uri']">
+                       # <xsl:value-of select="//*[@name='dc']/*[@name='relation']/*[@name='uri']" />
+		   </xsl:if>
+	        </field>
+            </xsl:if>
+	    <xsl:if test="//*[@name='description']/*[@name='abstract']/*/*[@name='value']">
+               <field name="description">
+		       <xsl:for-each select="//*[@name='description']/*[@name='abstract']/*/*[@name='value']">
+                     <xsl:value-of select="concat(., '&#10;')"/>
+                  </xsl:for-each>
+               </field>
+            </xsl:if>
+	    <!-- LANGUAGE -->
+            <xsl:for-each select="//*[@name='language']/*[@name='iso']">
+                <xsl:if test="string-length() > 0">
+                    <field name="language">
+                        <xsl:value-of select="php:function('VuFind::mapString', normalize-space(string(.)), 'language_map_iso639-1.properties')"/>
+                    </field>
+                </xsl:if>
+            </xsl:for-each>
+	    <!-- SUBJECT -->
+            <xsl:for-each select="//*[@name='subject']/*/*[@name='value']">
+               <field name="topic">
+                  <xsl:value-of select="normalize-space(.)"/>
+               </field>
+               <field name="topic_facet">
+                  <xsl:value-of select="normalize-space(.)"/>
+               </field>
+            </xsl:for-each>
+            <!-- Type -->
+	    <xsl:if test="//*[@name='type']">
+               <field name="format">
+		   <xsl:value-of select="//*[@name='type']"/>
+               </field>
+            </xsl:if>
+	    <!-- Rights - stored in the rights solr field for want of a better place -->
```

Comment:
This says "stored in the rights" but I think you mean "edition." However, my suggestion would be to use a dynamic field for this, like `rights_str`. Then you're not putting unexpected values into an existing field, but you make the data available to custom code.

Reply:
Thanks this is in my follow-up.